blog/config.md
---
tag: [主题配置,配置项,配置总览]
stick: true
---



# 所有配置

::: tip

主题从V1.3.2版本之后，就对页面进行重大改变，将主题配置移到`docs/.vuepress/config.js`内，并且加入了多种功能，如自动生成文章目录，侧边栏等效果...

`以下配置，没有特殊说明都是在docs/.vuepress/config.js ---> themeConfig内进行配置`



- 必读

> [主题问题解决](../issue/)
>
> [部署](../home/deploy.md)

- 推荐阅读

> [主题增加新组件，新页面](../feature/registercom.md)
>
> [修改主题默认样式](../style.md)
>
> [主题目录结构](../home/directory-structure.md)
>
> [主题写作技巧](../learn/markdown.md)
>
> [主题bug](../issue/bug.md)
>
> [更新日志](../issue/CHANGELOG.md)

- 环境安装

> [node安装教程](../node.md)

:::



## V1.8.xxx版本增加配置新版本<Badge type="tip" text="v1.8.xxx" vertical="top" />



### showRecommend

- Boolean

> 是否在文章页面展示文章推荐 默认展示





### articlePagination

- Boolean

> 是否在文章页面展示下一页，上一页，默认展示



### excludeTag<Badge type="tip" text="v1.8.0已被移除" vertical="top" />

> 从v1.8.0之后，被移除，改用`excludePath`



### excludePath

- Array

> 在统计文章，标签，类别的时候，需要排除的路径，只针对于根目录下的路径
>
> 主题内置已经排除了以下path
>
> `['/','/about/','/mood/','/link/','/tag/','/archive/','/photo/','/aurora-coze/','/aurora-register/','/aurora-archive/','/aurora-music/','/404.html']`，所以你如果想要排除的path在上面有，那么就不需要加入



::: details 点击查看图片

![image-20211122080443328](https://ooszy.cco.vin/img/blog-note/image-20211122080443328.png?x-oss-process=style/pictureProcess1)

:::



假如目前配置的为`excludePath: ['/footer.html','/v1.3.0/','/plugin/']`，文档的目录是上图那样，那么在根路径下(`根路径就是md文件直接放在docs目录下`)的`footer.md`会被直接排除，你不会在首页和tag页面看到该文章



::: tip

因为该排出是通过准确匹配文章的`URI`进行排除的，只能排除通过`localhost:[8080]/xxxx.html`或者`localhost:[8080]/v1.3.0/`进行访问的md文件

> 如果你有一个`README.md`文件，位置为`docs/v1.3.0/README.md`或者`docs/v1.3.0/readme.md`，那么这个`readme.md`文件可以直接`/v1.3.0/`或者`/v1.3.0/readme.html`(`会被重定向到/v1.3.0/`)或者`/v1.3.0/index.html`(`会被重定向到/v1.3.0/`)进行访问，所以该配置项只支持排除放在`docs`目录下或者在`docs`的子文件夹中的`readme.md`或者`README.md`文件

:::



<span style="color: red">记住所有文件名为`readme.md`或者`README.md`的文件，都可以直接通过`XXX/`进行访问</span>

> 这里的`XXX/`并不是你在浏览器中输入`localhost:[8080]/XXX/`....





### wave



### showWave

- Boolean

> 是否在首页展示波浪效果



::: details code

```js
wave: {
    showWave: true
}
```

![image-20211122081653795](https://ooszy.cco.vin/img/blog-note/image-20211122081653795.png?x-oss-process=style/pictureProcess1)

:::



### bubble

- Object

> 主题气泡配置项，该项只会影响顶部图片那里的气泡

:::  details view picture

![image-20211122081842634](https://ooszy.cco.vin/img/blog-note/image-20211122081842634.png?x-oss-process=style/pictureProcess1)

:::



### show<Badge type="tip" text="bubble下的配置项" vertical="top" />

- Boolean

> 是否在顶部图片中显示气泡效果







### bubbleNumber<Badge type="tip" text="推荐0-1之间小数,不能为0" vertical="top" />

- Number

> 气泡数量 推荐0(不包括)到1之前的小数，



### bubbleAlpha<Badge type="tip" text="推荐0-1之间小数,不能为0" vertical="top" />

- Number

> 气泡透明度 0到1之间的小数




### alphaChangeSpeed<Badge type="tip" text="越接近于0越好,不能为0" vertical="top" />
- Number

> 透明度变化速度，越接近于0越好




### size<Badge type="tip" text="推荐0-1之间小数,不能为0" vertical="top" />
- Number

> 气泡大小，推荐0到1之间的值




### sizeChangeSpeed<Badge type="tip" text="越接近于0越好,不能为0" vertical="top" />
- Number

> 气泡大小变化速度 越小越好
> 


### riseSpeed<Badge type="tip" text="推荐0-1之间小数,不能为0" vertical="top" />

- Number

> 气泡上升速度



### color

- String

> 气泡颜色，如果你想要设置为白色`rgb(255,255,255)`，那么这里需要传入请传入`255,255,255`



## 新版本<Badge type="tip" text="v1.3.2-v1.6.2" vertical="top" />

::: tip 

这部分的配置是从v1.3.2到v1.6.2版本新增或者修改的配置项

:::

### sidebarAvatar<badge text="v1.6.2"/>

- String

> 侧边栏头像，如果此项为undefine,那么将使用heroImg

### navbar<badge text="配置项被修改过"/>

::: details 原来

```js
{
    text: '快速开始',
    link: '/readme/',
},
{
    text: "所有配置",
    link: '/home/config.html',
},
```

:::



::: details 现在

```js
{
    text: '快速开始',
    link: '/readme/',
    iconClass: 'aurora-0023pacman'
},
{
    text: "所有配置",
    link: '/home/config.html',
    iconClass: 'aurora-link'
},
```

:::

> 新增自定义顶部导航栏icon图标

自定义导航栏的图标请查看[自定义图标](../style/icon.md)





### about<badge text="增加对html的支持"/>

关于页面

```js
about: [
    {
        bar: false,
        title: "我?",
        describe: [
            "这是关于页面支持html<a href='https://blog.cco.vin' target='_blank'>这是一个标签</a>",
            "目前是一名大三学生,CS专业,坐标西南边陲"
        ],
    }
    }
```

::: tip

可以在describe中，写入html

:::



### friendLinks<badge text="发生重大修改，影响运行"/>

::: details 以前

```js
friendLinks: [
    {
        //网站标题
        title: "XI溪",

        //站点链接
        url: "http://www.xiaoxuya.top",

        //站点logo
        logo: "https://www.xiaoxuya.top/img/logo.png",

        //站点描述
        describe: "人生若只是初见,何事秋风悲画扇",
    },
    {
        title: "左眼会陪右眼哭の博客",
        url: "http://qkongtao.cn/",
        logo: "http://qiniu.qkongtao.cn/2020/12/d11-e1628358435552.png",
        describe: "干嘛这么想不开，要在脸上贴个输字！",
    }
],

```

:::



::: details 现在

```js
friendLinks: [
    {
        title: '我的朋友',
        links: [
            {
                title: "xxx",
                url: "xxx",
                logo: "xxx",
                describe: "xxx",
                cover: 'xxx'
            }
        ]
    },
    {
        title: '大佬',
        links: [
            {
                title: "xxx",
                url: "xxx",
                logo: "xxx",
                describe: "xxx",
                cover: 'xxx'
            }
        ]
    },
    {
        title: '小伙伴',
        links: [
            {
                title: "xxx",
                url: "xxx",
                logo: "xxx",
                describe: "xxx",
                cover: 'xxx'
            },
        ]
    },
],
```

:::

::: tip

以前友情链接页面，只能配置一个模块，也就是我的朋友，现在支持配置多个模块，`friendLinks`是一个数组对象，效果请查看配置之后的效果，友情链接地址`/link`

:::



### cover<badge text="v1.6.0"/>

- String

> 配置友情链接的封面，如果没有指定，那么默认是白色背景



::: details view

```js
{
    title: "xxx",
    url: "xxx",
    logo: "xxx",
    describe: "xxx",
    cover: 'https://ooszy.cco.vin/img/blog-note/image-20211111084654046.png?x-oss-process=style/pictureProcess1'
}
```

:::



### siteInformation<badge text="v1.6.0 发生更改"/>

::: details 以前

```js
siteInformation: {
    //站点标题
    title: "qsyyke",

    //自己站点链接
    url: "https://www.cco.vin",

    //自己站点logo
    logo: "https://ooszy.cco.vin/img/blog-public/avatar.jpg",

    //自己站点描述
    describe: "I do not follow,i lives is always all you want",

    //自己的头像
    email: "cqycco@gmail.com",
},
```

:::



::: details 现在

```js
siteInformation: {
    //站点标题
    title: "Aurora-theme",

    //自己站点链接
    url: "https://aurora.cco.vin",

    //自己站点logo
    logo: "https://ooszy.cco.vin/img/ico/yuan.png",

    //自己站点描述
    describe: "vuepress-theme-Aurora是一款简洁，美观，功能强大的静态主题",
    cover: 'https://ooszy.cco.vin/img/blog-note/image-20211106100103898.png?x-oss-process=style/pictureProcess1',

    //自己的头像
    contact: "email: cqycco@gmail.com, qq:2291308094",
    otherDescribe: [
        '申请友链请按照下面格式，在此页面留言，我看到就会进行添加',
        '博客名称    博客地址      博客描述      博客logo',
        '友链申请要求',
        '拒绝涉政/涉黄/太多广告/盈利性站点',
        '站点能正常访问',
        '这是一个html的测试<a href="https://blog.cco.vin" target="_blank">这是一个a标签</a>'
    ]
},
```



:::





### contact<badge text="v1.6.0 新增"/>

- String

> 在自己的站点信息处，添加联系方式如
>
> - `contact: "email: cqycco@gmail.com, qq:2291308094"`
>
>   > 展示为`//email: cqycco@gmail.com, qq:2291308094`
>
> - `contact: "如果几天我还未添加你站点，请联系我qq2291308094"`
>
>   > 展示为`//如果几天我还未添加你站点，请联系我qq2291308094`



::: details 展示效果

![image-20211111090308201](https://ooszy.cco.vin/img/blog-note/image-20211111090308201.png?x-oss-process=style/pictureProcess1)

:::

### otherDescribe<badge text="v1.6.0 新增"/>

- Array

> 自己站点的其他描述，主要是添加一些添加友联的要求，里面的每一个都支持html

::: details view

```js
siteInformation: {
    //站点标题
    title: "Aurora-theme",

    //自己站点链接
    url: "https://aurora.cco.vin",

    //自己站点logo
    logo: "https://ooszy.cco.vin/img/ico/yuan.png",

    //自己站点描述
    describe: "vuepress-theme-Aurora是一款简洁，美观，功能强大的静态主题",
    cover: 'https://ooszy.cco.vin/img/blog-note/image-20211106100103898.png?x-oss-process=style/pictureProcess1',

    //自己的头像
    contact: "email: cqycco@gmail.com, qq:2291308094",
    otherDescribe: [
        '申请友链请按照下面格式，在此页面留言，我看到就会进行添加',
        '博客名称    博客地址      博客描述      博客logo',
        '友链申请要求',
        '拒绝涉政/涉黄/太多广告/盈利性站点',
        '站点能正常访问',
        '这是一个html的测试<a href="https://blog.cco.vin" target="_blank">这是一个a标签</a>'
    ]
},
```

:::



### message<badge text="v1.6.0 html支持"/>

```js
message: [
    '该博客主题为Aurora,<a href="https://github.com/qsyyke/vuepress-theme-aurora">vuepress-theme-Aurora</a>',
    "主题交流群: 681602026，欢迎各位大佬进群交流",
]
```



### showMoodEdit<badge text="v1.6.0 被移除"/>



### addMood<badge text="v1.6.0 被移除"/>



### showOnlineMood<badge text="v1.6.0 被移除"/>



### articleH2Icon<badge text="v1.6.0 新增"/>

- String

> 二级标签的前面图标，默认`🌸`,请从[emoji-cheat-sheet](https://github.com/ikatyang/emoji-cheat-sheet)直接复制



### articleH3Icon<badge text="v1.6.0 新增"/>

- String

> 三级标签的前面图标，默认`🐳`,请从[emoji-cheat-sheet](https://github.com/ikatyang/emoji-cheat-sheet)直接复制

###  articleH4Icon<badge text="v1.6.0 新增"/>

- String

> 四级标签以四级标签之后的前面图标，默认`⛄`,请从[emoji-cheat-sheet](https://github.com/ikatyang/emoji-cheat-sheet)直接复制



::: details 展示效果

![image-20211111090023642](https://ooszy.cco.vin/img/blog-note/image-20211111090023642.png?x-oss-process=style/pictureProcess1)

:::

## 首页

<a href="https://aurora.xcye.xyz/config/">点击查看配置</a>

### home

- Boolean

> true为首页，false反之
>
> `此项在docs/readme.md`内配置，用于设置首页位置



### logoColor

- String 

> 首页左上角Logo图旁文字颜色



### logoTitle

- String

> 左上角，logo旁标题文字
> 


### defaultBorderRadius

- Number

> 默认的圆角，传入数字
> 

### defaultOpacity

- Number

> 默认的透明度，传入0到1之间的小数，0表示全透明
> 

### isHomePageFollow

- boolean

> 首页文章列表透明度是否跟随样式面板改变，true表示跟随，全白色,false表示不跟随
> 

### defaultBlur

- Number

> 默认模糊度(`0 到 1之间`)

### showFont

- String

> 圆角等控制面板，字体切换的占位符文字，推荐一个字



### faviconIco

- String

> 网站顶部favicon图片链接，如https://ooszy.cco.vin/img/blog-public/ccds_64.ico
>
> `v1.3.2移除，改在config.js的head内配置`

::: details 实例

```js
module.exports = {
    ...
    head: [
        [
            "link",
            {
                href: "https://ooszy.cco.vin/img/ico/yuan.png",
                rel: "icon",
            },
        ],
    ],
}
```

:::



### customRandomSay

- Boolean

> 是否自定义随机一言，true表示不会使用随机一言接口
>
> 默认使用的接口为:https://international.v1.hitokoto.cn/?c=b&max_length=45，感谢大佬提供的一言接口



### customRandomValue

- String

> 自定义一言文字，只有customRandomSay为true时有效



### showHomeMood

- Boolean

> 是否在首页显示心情，true开启，需要设置文字
>
> `v1.3.2已移除`

### mood

- String

> 首页中间，心情文字
>
> `v1.3.2已移除`





### heroLogo

- String

> 首页中间图片链接
>
> `v1.3.2已移除`



### heroImg

- String

> 首页中间图片链接，默认https://ooszy.cco.vin/img/blog-public/avatar.jpg
>



### logo

- String

> 左上角logo图片



### cancelIcoHref

- String

> 圆角字体控制面板，取消按钮图片链接
>
> `v1.3.2已移除，改用阿里字体图标，如果修改，请f12，修改对应值`，组件位置为docs/.vuepress/theme/lib/client/components/child/home/StyleMenu.vue



### description

- String 

> 首页和非文章页面的description描述文字



### keyword

- String

> 首页和非文章页面的keyword，请使用`,`分割开





### slideTime

- Number

> 圆角控制面板，隐藏时间，单位毫秒



### randomSaw

- String

> 随机一言接口 https://international.v1.hitokoto.cn/?c=b&max_length=45


### randomSawQuery

- String

> 从随机一言接口中，如何获取值，如上面`randomSaw`接口，输入hitokoto(`后续版本会优化此项`)

### method

- String

> 随机一言请求接口方法



animeOption: {
  baseURL: https://api.vvhan.com/api/acgimg?type=json,
  method: get,
  timeout: 5000,
  query: imgurl
}



blog/.vuepress/config.js
const { path } = require("@vuepress/utils");
const {navbar} = require("./navbar.js")
const {friendLinks} = require("./friendLinks")
const {socials} = require("./socials");
const {randomColor} = require("./randomColor");
const {siteInformation} = require("./siteInformation");
const {message} = require("./message");
const {donate} = require("./donate");
module.exports = {
    base: '/qhjack-blog/',//如果你准备发布到 https://<USERNAME>.github.io/<REPO>/ ，也就是说你的仓库地址是 https://github.com/<USERNAME>/<REPO> ，则将 base 设置为 "/<REPO>/"。

    /*
    * 这里是配置Markdown语法增强的配置，如需配置，请查看官方介绍https://v2.vuepress.vuejs.org/zh/guide/markdown.html
    * 或者你也可以看过我的配置，使用该配置的文件路径Aurora-theme/lib/node/auroraTheme.js
    * */
    extendsMarkdown: (md) => {

    },

    // 打开下面注释，将使用webpack作为打包工具，如果注释将使用vite作为打包工具[目前主题不能使用vite打包，会报错]
    bundler: '@vuepress/bundler-webpack',
    bundlerConfig: {
    },

    /*
    * 在这里配置插件
    * 插件plugins是一个数组，官方介绍: https://v2.vuepress.vuejs.org/zh/guide/plugin.html
    * */
    plugins: [
        "vuepress-plugin-comment2",
        {
          type: "giscus",
          repo: "jack9603301/qhjack-blog",
          repoId: "R_kgDOHVZ-oQ",
          category: "General",
          categoryId: "DIC_kwDOHVZ-oc4CPGa5",
          mapping: "pathname"
        },
        "@vuepress/plugin-search",
        {
            locales: {
                "/": {
                    placeholder: "Search",
                },
                "/zh/": {
                    placeholder: "搜索",
                },
            },
        },
        /*
        * 在主题中，使用path.resolve(__dirname,的地方，都是使用本地的插件或者主题，看位置
        * 这里是配置说说插件，使用https://aurora.xcye.xyz/plugin/coze/
        * */
        [
            "coze",
            {
                appId: '2A2Dyd2Affrnldhwftlaaa',//你需要在leancloud中创建自己的appId，请参照文档https://aurora.xcye.xyz/plugin/coze/
                appKey: 'qHYTbb91iOPLelyC9aaa',
                masterKey: 'eUwfvS3luIPnPiaaa',
                //下面这些是可选的
                avatarPath: '/avatar.jpg',//说说头像url
                registerPath: '/register', //自定义插件默认提供的注册页面路由，请在前面加上/
                onlyAdministrator: false //是否运行其他注册的用户发布说说，true表示只有管理员可以发布
            }
        ],
        /*
        * 该插件是配置时间轴，使用https://aurora.xcye.xyz/plugin/archive/
        * 请配置一下excludes，否则在时间轴那里，会出现非文件页面链接
        * */
        [
            "archive",
            {
                excludes: ['/footer.html','/404.html','/about/','/mood/','/link/','/tag/','/photo/'],
                noTitle: '暂时没有标题配置'
            }
        ],

        /*
        * 气泡插件，使用：https://aurora.xcye.xyz/plugin/bubble/
        * */
        [
            "bubble",
            {
                //气泡数量 推荐0(不包括)到1之前的小数，
                bubbleNumber: 0.14,

                //气泡透明度 0到1之间的小数
                bubbleAlpha: 0.6,

                //透明度变化速度，越接近于0越好
                alphaChangeSpeed: 0.00001,

                //气泡大小，推荐0到1之间的值
                size: 0.4,

                //气泡大小变化速度 越小越好
                sizeChangeSpeed: 0.0002,

                //气泡上升速度
                riseSpeed: 0.4,

                //气泡颜色，白色rgb(255,255,255) 请传入255,255,255
                color: '255,255,255',

                zIndex: -2
            }
        ],
    ],
    /*
    * 设置head 一定要加入<script src="https://at.alicdn.com/t/font_2849934_v6y652peian.js"></script>项配置，否则一些图标不能正常显示
    * 如果你需要额外引入css样式或者是js文件，请在head中引入，如何引入，可以参照下面的引入方式
    * */
    head: [
        [
            "script",
            {
                src: "https://at.alicdn.com/t/font_2849934_v6y652peian.js",
            },
        ],
        //设置站点icon
        [
            "link",
            {
                href: "/avatar.png",
                rel: "icon",
            },
        ],
        [
            "link",
            {
                rel: 'stylesheet',
                type: 'text/css',
                href: '//at.alicdn.com/t/font_2932340_r7zitafg82.css'
            }
        ],
        /*[
            "link",
            {
                rel: 'stylesheet',
                type: 'text/css',
                href: '//at.alicdn.com/t/font_2951154_btu3y5blqnn.css'
            }
        ]*/
    ],
    //这里使用本地主题
    //theme: path.resolve(__dirname, "../../Aurora-theme/lib/node/index.js"),
    theme: 'aurora',

    //站点title
    title: "起航天空，引领我们的未来",

    //设置站点语言
    lang: 'zh-CN',

    //设置运行npm run dev的端口
    port: 8080,

    /*
    * themeConfig是和主题相关的配置项，主题的所有配置，都是在此处进行配置的
    * */
    themeConfig: {
        /*
        * 这里配置导航栏项，建议将navbar提取出，比如navbar.js
        * 使用：https://aurora.xcye.xyz/homeconfig.html#%E5%AF%BC%E8%88%AA%E6%A0%8F
        * 因为navbar会存在很多的配置，所以推荐在该目录下，创建一个navbar.js文件，此文件中，使用module.exports = {navbar: XXX}
        * 可以参照docs/.vuepress/navbar.js使用，然后在config.js的themeConfig处引入该文件
        * */
        navbar: navbar,

        //禁用黑夜模式，当前版本未提供
        darkMode: false,

        //项目地址，会在顶部导航栏和侧边栏显示GitHub的地址
        repo: "https://gitee.com/jack960330/qhjack-blog",

        //自定义GitHub的图标
        repoIconClass: 'aurora-navbar-github',

        //md文件的仓库地址
        docsRepo: 'https://gitee.com/jack960330/qhjack-blog',

        /*
        * 站点是否是使用github Actions自动部署，如果你使用github Actions完成自动部署，那么这里一定要为TRUE，为TRUE，假如你把此项目push
        * 到https://github.com/qsyyke/aurora-docs这个仓库中，那么你需要将docsRepo设置为https://github.com/qsyyke/aurora-docs，
        * 那么在文章页面，会根据根据你docsRepo和docsBranch设置正确的edit链接，也就是可以一键打开GitHub仓库中，此篇md文件的编辑
        * */
        githubActions: false,

        //md文件存放的仓库分支
        docsBranch: 'develop',

        //顶部导航栏，你repo地址的超链接文本
        repoLabel: "qhjack-blog",

        //是否启用文章在线编辑，默认开启，会在每篇文章的底部，创建一个链接，指向该md在GitHub中的edit地址
        editLink: true,

        //在线编辑文字，点击编辑该文章的label
        editLinkText: "edit",

        //该md最后更新时间，使用github commit的提交时间
        lastUpdated: true,
        lastUpdatedText: "lastTime",

        //下面的都是主题自己的配置文件
        //logo旁文字颜色
        logoColor: "#2c3e50",

        //样式控制面板字体占位符，如果是国内用户，请使用中文
        showFont: "程",

        //首页中间hero图片地址
        heroImg: '/avatar.jpg',

        //logo图片地址
        logo: "/avatar.png",

        //文章懒加载图片 仅限文章，首页文章占位图片并不是这个
        lazyLoadingImg: "/ljz.gif",

        //这是首页文章列表懒加载图片
        homePageLazyLoadingImg: '/aurora-loading.gif',

        //是否启用定制首页随机一言，默认未开启，使用随机一言 接口为https://international.v1.hitokoto.cn/?c=b&max_length=45
        customRandomSay: true,

        //定制首页随机一言文字
        customRandomValue: '在茫茫之中，探索自然奥秘，在星星之源，绽放声明之花',

        //社交信息，首页PC端至多显示19个，手机端至多显示7个，侧边栏不影响，推荐自己创建一个socials.js
        socials: socials,

        //logo旁文字 默认值为Aurora
        logoTitle: "起航天空",

        //貌似没用的配置项┭┮﹏┭┮
        headTitle: "this is headTitle",

        //站点描述
        description: "在茫茫之中，探索自然奥秘，在星星之源，绽放声明之花。起航天空，引领你的未来，配棒你的一生！",

        //站点关键词，在后续版本中，还需优化 请使用英文状态下的逗号','隔开
        keyword: "起航天空,博客,起航,天空",

        //样式控制面板打开之后，休眠多长时间自动关闭面板，单位毫秒
        slideTime: 300000,

        //随机一言接口 没用配置
        randomSaw: "https://international.v1.hitokoto.cn/?c=b&max_length=45",
        /*
        * 关于页面 推荐单独创建一个about.js文件存放关于页面信息
        * 使用：https://aurora.xcye.xyz/homeconfig.html
        * */
        about: [
            {
                bar: false,
                title: "我?",
                describe: [
                    "我是一名喜欢开源技术的普通人，涉足智能家居，3d打印等，坐标广州"
                ],
                tag: [
                    "coding",
                    "宅男",
                    "动画",
                    "社控",
                    "懒",
                ],
                showTag: true,
            },
        ],

        //这是配置随机背景；颜色 可以不设置，有默认值 推荐单独创建一个random.js文件
        randomColor: randomColor,

        //样式控制面板至多显示多少个字体和字体颜色，推荐不超过8个
        maxFontColorArr: 8,

        //在样式控制面板中，显示的字体颜色集合
        fontColor: [
            "#2c3e50", "#42a5f5", "#8093f1", "#FF6EC7", "#FF7F00", "#8FBC8F", "#EAADEA",
            "#3299CC", "#CDCDCD", "#CC3299", "#FF7F00", "#2F4F4F",
        ],

        /*
        * 友情链接数组 推荐单独创建一个friendLinks.js存放友情链接的配置，可以参照navbar项的配置
        * 使用：https://aurora.xcye.xyz/page/friendlink.html
        * */
        friendLinks: friendLinks,

        //自己的站点信息 我自己的站点描述 会显示在友情链接的底部，推荐单独创建一个文件存放
        siteInformation: siteInformation,

        //需要排除的标签，自动生成的标签中，不会有这个标签，这是一个数组
        //excludeTag: ["note"],

        //样式控制面板显示的字体，有默认值
        fontFamily: [
            "-apple-system", "hlt", "tzt", "sst", "lf", "xsf", "lsf", "cgt",
        ],

        //页脚信息，支持HTML，这是一个数组
        footer: [
            "Copyright © 起航天空.",
        ],

        //是否显示页脚，控制全局
        isShowFooter: true,

        //是否显示主题信息在页脚，为false关闭
        isShowThemeCopyright: false,

        //是否展示运行时间
        isShowRunTime: true,

        //网站开始时间，请按照以下格式进行
        startRunTime: "5/13/2022 17：51：50",

        //网站运行时间前缀
        prefixRuntime: "起航天空已运行",

        //公告，是一个数组，支持图片，HTML
        message: message,

        //文章底部最大推荐文章数 默认值为30
        recommendPageLength: 30,

        //推荐列表标题为空时，就会使用这个进行代替，默认是`╮(￣▽￣)╭`
        recommendNoTitle: "`╮(￣▽￣)╭`",

        //tag页，没有标题时，代替文字 默认是下面这个
        tagNoTitle: "暂时还没有标题哟",

        //首页中间框的话语
        mood: "起航天空",

        //默认打开网站时的毛玻璃状态，TRUE表示默认开启毛玻璃效果
        isFitter: true,

        //默认的圆角，传入数字
        defaultBorderRadius: 10,

        //默认的透明度，传入0到1之间的小数，0表示全透明
        defaultOpacity: 1,

        //首页文章列表透明度是否跟随样式面板改变，true表示跟随，全白色,false表示不跟随
        isHomePageFollow: false,

        //默认模糊度
        defaultBlur: 1,

        //手机端，是否在页面的底部显示侧边栏列表，默认开启，如果需要开启，请将此值设置为FALSE
        //mobilePageSidebar: false,

        //tag页面，标签分割符 请不要传入一个空字符串，默认值就是' ' 一个空格
        split: "~",

        //赞赏信息
        donate: donate,

        //海报分享配置
        poster: {
            //博客描述
            description: "在茫茫之中，探索自然奥秘，在星星之源，绽放声明之花。起航天空，引领你的未来，配棒你的一生！",

            //作者
            author: "起航天空",

            //博客前缀
            preBlog: "qhjack-blog",

            //海报博客名称后缀
            suffixBlog: " theme",

            //头像，请放置在docs/public目录下，或者请保证此图片链接能够跨域访问，否则头像不能正常显示
            avatar: "/avatar.jpg",
        },

        //是否展示从网络上请求回来的说说 如果启用，请自己写后台服务，修改源码，目前暂未在主题中加入，期望在将来能够实现，组件位置docs/.vuepress/theme/lib/client/components/Mood.vue
        showOnlineMood: false,

        //是否开启在线添加说说功能，如果需要，请自己写后台，修改源码，目前在主题内部暂未加入 组件位置docs/.vuepress/theme/lib/client/components/child/AddMood.vue
        showMoodEdit: false,

        //自定义顶部图片
        customTopImg: {
            //是否启用定制顶部图片，控制全局，如果关闭，那么将使用随机图片，随机图片接口可以自己设置
            custom: false,

            //文章顶部图片，数组，每次从数组中随机选择一张
            page: [
                "/bg/555260.png",
                '/bg/404901.png',
                '/bg/734386.png'
            ],
            //友情链接页面
            friend: [
                "/bg/669.png",
                '/bg/5332.png'
            ],
            //标签页面
            tag: [
                "/bg/763311.png"
            ],
            //心情页面
            mood: [
                "/bg/5849.png"
            ],
        },

        //首页背景图片数组，考虑到使用随机图片，打开网站速度变慢，所以移除随机图片，使用自己设置的图片链接
        homeWps: [
            "/bg/1.png",
            '/bg/4.png',
        ],

        //手机端首页背景图片
        homeWpsMobile: [
            "/bg/1.png",
            '/bg/4.png',
        ],

        //首页文章显示条数，默认为4，此值不推荐设置太大
        pageSize: 4,

        //侧边栏配置
        //github地址
        githubUrl: "https://gitee.com/jack960330/qhjack-blog",

        //最新文章数量，默认为6
        latestPageSize: 6,

        //首页是否显示文章图片，默认关闭，如果显示的话，首页加载会非常慢 已弃用，从v1.3.2开始，直接移除首页文章内容图片
        showHomePageImg: false,

        //文章侧边栏自动获取的层次 默认为1，也就是http://localhost:8080/config/feature/donate.html,只会自动生成feature目录下的文件
        sidebarCatalogLevel: 3,

        //首页文章列表封面图api接口
        homePageImgApi: "https://www.dmoe.cc/random.php?return=text?time=-Infinity",

        //手机端侧边栏横线分割文字，默认为Aurora
        mobileCutText: "起航天空",

        //侧边栏标签处显示还是分类还是标签，只有两个值，默认为分类，如果为categories，那么就显示为类别，否则显示为标签
        sidebarTag: "分类",

        //额外的功能，也就是样式控制面板上面那个，默认是关闭的
        showAddMood: true,

        //生成海报的顶部图片api接口，请注意，该接口需要直接返回图片地址，不能有跨域问题，设置之前，可以先使用ajax看是否存在跨域
        postImgApi: 'https://picture.cco.vin/pic/rmimg?type=bing',

        /*
        * 以下为v1.5.4新增功能
        * */

        //文章h2标签的icon
        articleH2Icon: '🌸',

        //文章h3标签的icon
        articleH3Icon: '🐳',

        //文章h4标签及h4标签之后的标签的icon
        articleH4Icon: '⛄',

        //v1.6.4新增配置
        sidebarAvatar: '/avatar.jpg',

        //1.7.1增加配置
        sugCountPerMin: 300,

        //v1.8.0新增配置项
        //是否展示文章推荐 默认展示
        showRecommend: true,

        //文章页面是否显示上一页，下一页 默认展示
        articlePagination: true,

        //这个global暂时没用
        global: {
            sidebar: {
                showStatus: true,
                perData: false,
                social: false,
                friendLink: true,
                navbar: false,
                latestPage: true,
                message: false,
                tag: true,
                siteData: false
            },
            footer: false,
        },

        //excludeTag: ["note"], 此配置项，不再需要使用

        //在统计文章，标签，类别的时候，需要排除的路径，只针对于根目录下的路径，
        excludePath: ['/footer.html','/v1.3.0/','/plugin/',"/node.html","/style.html"],

        //类别项是否包含文件夹名，默认包含
        categoriesIncludeFolderName: false,
        //首页波浪效果设置
        wave: {
            showWave: true
        },

        //顶部图片的气泡控制
        bubble: {
            show: true,

            //气泡数量 推荐0(不包括)到1之前的小数，
            bubbleNumber: 0.14,

            //气泡透明度 0到1之间的小数
            bubbleAlpha: 0.6,

            //透明度变化速度，越接近于0越好
            alphaChangeSpeed: 0.00001,

            //气泡大小，推荐0到1之间的值
            size: 0.4,

            //气泡大小变化速度 越小越好
            sizeChangeSpeed: 0.0002,

            //气泡上升速度
            riseSpeed: 0.4,

            //气泡颜色，白色rgb(255,255,255) 请传入255,255,255
            color: '255,255,255'
        },

        /**
         * 这里是配置统计阅读量的leanCloud配置，从1.9.0开始，将不再需要依赖waline的阅读统计,这里配置的appId,appKey,masterKey可以
         * 和vuepress-plugin-coze插件使用同一个应用
         * */
        leanCloud: {
            appId: '2A2Dyd2AffrnldhwftlEddVn-MdYXbMMI',
            appKey: 'qHYTbb91iOPLelyC9lpbXxLH',
            masterKey: 'eUwfvS3luIPnPiHS5SpEhDYr',
        },


        //这是v1.9.0新增加的功能 是否在文章页底部显示最后更新时间，贡献者，点击编辑,默认显示
        showPageMeta: true,

        /*
        * 以下是1.11.0版本新增的配置项
        * */

        //随机一言接口，请注意，一定要保证该接口直接返回Text文本，一定要保证该接口直接返回Text文本，而不是返回json
        randomSayApi: {
            method: 'GET',
            urlApi: 'https://v1.hitokoto.cn/?encode=text&c=a'
        },

        /*
        * 一下是v1.11.1版本增加的配置
        * */
        afDianUrl: 'https://afdian.net/@qsyyke',//你的爱发电个人页面地址

        /*
        * 下面这些是v1.12.0版本新增的配置项
        * */

        // 首页置顶文字
        homeTopText: '置顶',

        //侧边栏一句话描述
        sidebarDesc: '在茫茫之中，探索自然奥秘，在星星之源，绽放声明之花。起航天空，引领你的未来，配棒你的一生！',

        /*
        * 下面的是v1.13.0版本的配置
        * */

        /*
        * 如果你以前使用的是其他的主题或者工具，比如hexo的主题，reco主题等等，发现frontmatter中的标签并不是使用tag字段设置，而是使用比如tags进行设置的
        * 那么你可以像customTagName: 'tags'自定义标签的字段，此配置仅仅针对于md文件设置的标签并不是使用tag字段，请注意，谨慎使用
        * */
        //customTagName: 'tags',

        /*
       * 如果你以前使用的是其他的主题，比如hexo，reco等等，发现frontmatter中的类别并不是使用categories字段，而是使用比如category进行设置的
       * 那么你可以像customCategoriesName: "category"这样自定义类别的字段，此配置仅仅针对于md文件设置的类别并不是使用categories字段，请注意，谨慎使用
       * */
        //customCategoriesName: "category",

        /*
        * 下面这个是v1.13版本新增的配置项
        * */

        //是否显示关于页面气泡效果 默认开启
        showAboutPageBubble: true,
    }
};
